### ISSUE TITLE
Sampling edge connection according to distance on MPNN

### ISSUE THREAD

**Issue by Mario-Liu** [rusty1s/pytorch_geometric#1828]
I want to determine the node-node connection in MPNN based on the distance between nodes (rather than graph full connection), for example, when the node distance is less than 3, there are connected edges between nodes for message passing.  How can I get the distance according to  the "edge_index"   defined by "data.edge_index = edge_index".  Can you show me what to do?

class MyTransform(object):
    def __call__(self, data):
        # Specify target.
        data.y = data.y[:, target]
        return data

class Complete(object):
    def __call__(self, data):
        device = data.edge_index.device

        row = torch.arange(data.num_nodes, dtype=torch.long, device=device)
        col = torch.arange(data.num_nodes, dtype=torch.long, device=device)

        row = row.view(-1, 1).repeat(1, data.num_nodes).view(-1)
        col = col.repeat(data.num_nodes)
        edge_index = torch.stack([row, col], dim=0)

        edge_attr = None
        if data.edge_attr is not None:
            idx = data.edge_index[0] * data.num_nodes + data.edge_index[1]
            size = list(data.edge_attr.size())
            size[0] = data.num_nodes * data.num_nodes
            edge_attr = data.edge_attr.new_zeros(size)
            edge_attr[idx] = data.edge_attr

        edge_index, edge_attr = remove_self_loops(edge_index, edge_attr)
        **data.edge_attr = edge_attr
        data.edge_index = edge_index**

        return data

path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'QM9')
transform = T.Compose([MyTransform(), Complete(), T.Distance(norm=False)])
dataset = QM9(path, transform=transform).shuffle()

**Comment by rusty1s** [comment]
You can use the `torch_geometric.transforms.RadiusGraph` transform to achieve that, *e.g.*:
```python
import torch_geometric.transforms as T
dataset = QM9(path, transform=T.RadiusGraph(r=...)
```
In case you are only interested in getting the distance between edges in `edge_index`, you can compute them via:
```python
row, col = edge_index
(pos[row] - pos[col]).norm(dim=-1)
```

**Comment by Mario-Liu** [comment]
> 
> 
> You can use the `torch_geometric.transforms.RadiusGraph` transform to achieve that, _e.g._:
> 
> ```python
> import torch_geometric.transforms as T
> dataset = QM9(path, transform=T.RadiusGraph(r=...)
> ```
> 
> In case you are only interested in getting the distance between edges in `edge_index`, you can compute them via:
> 
> ```python
> row, col = edge_index
> (pos[row] - pos[col]).norm(dim=-1)
> ```

Great！ Using "dataset = QM9(path, transform=T.RadiusGraph(r=...)"  is a great way!  BTW, if I'd like to not only include all nodes within the  radius, but also random sample part of the nodes that are outside the radius.  Can we still use any existing transforms directly int the way like ?

`import torch_geometric.transforms as T
dataset = QM9(path, transform=T.RadiusGraph[(all nodes r< radius)&(few nodes r> radius)]`
